igor-core/src/main/java/com/netflix/spinnaker/igor/polling/CommonPollingMonitor.java

```diff
@@ -163,7 +163,17 @@ protected String getLockName(String name, String partition) {
   }
 
   protected void internalPollSingle(PollContext ctx) {
-    String monitorName = getClass().getSimpleName();
+    String monitorName =
+        !StringUtils.isBlank(this.getName()) ? this.getName() : getClass().getSimpleName();
```

Comment:
I think this is was a bug, cause overridden function was totally ignored in metrics